Plugins/GRPCProtobufGenerator/Plugin.swift

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import PackagePlugin
+
+@main
+struct GRPCProtobufGenerator {
+  /// Build plugin code common to both invocation types: package manifest Xcode project
+  func createBuildCommands(
+    pluginWorkDirectory: URL,
+    tool: (String) throws -> PluginContext.Tool,
+    inputFiles: [URL],
+    configFiles: [URL],
+    targetName: String
+  ) throws -> [Command] {
+    let configs = try readConfigurationFiles(configFiles, pluginWorkDirectory: pluginWorkDirectory)
+
+    let protocGenGRPCSwiftPath = try tool("protoc-gen-grpc-swift").url
+    let protocGenSwiftPath = try tool("protoc-gen-swift").url
+
+    var commands: [Command] = []
+    for inputFile in inputFiles {
+      guard let config = configs.findApplicableConfig(for: inputFile) else {
+        throw PluginError.noConfigurationFilesFound
+      }
+
+      let protocPath = try deriveProtocPath(using: config, tool: tool)
+      let protoDirectoryPath = inputFile.deletingLastPathComponent()
+
+      // unless *explicitly* opted-out
+      if config.client != false || config.server != false {
```

Comment:
Now that these aren't optional we can be more idiomatic
```suggestion
      if config.client || config.server {
```